trace/redis/redis.go

```diff
@@ -0,0 +1,85 @@
+package redis
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/beatlabs/patron/trace"
+	"github.com/go-redis/redis/v7"
+	"github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/ext"
+)
+
+const (
+	// RedisComponent Definition
+	RedisComponent = "redis"
+	// RedisDBType description
+	RedisDBType = "In-memory"
+	versionTag  = "version"
+)
+
+var (
+	version = "dev"
+)
+
+// Span starts a new Redis child span with specified tags.
+func Span(ctx context.Context, opName, cmp, dbType, instance, stmt string,
+	tags ...opentracing.Tag) (opentracing.Span, context.Context) {
+
+	sp, ctx := opentracing.StartSpanFromContext(ctx, opName)
+	ext.Component.Set(sp, cmp)
+	ext.DBType.Set(sp, dbType)
+	ext.DBInstance.Set(sp, instance)
+	ext.DBStatement.Set(sp, stmt)
+	for _, t := range tags {
+		sp.SetTag(t.Key, t.Value)
+	}
+	sp.SetTag(versionTag, version)
```

Comment:
if we do not allow the version to be passed in setting it something fixed gives no value.